docs/interactions/Message_Components.md

```diff
@@ -418,3 +423,50 @@ Select menus support single-select and multi-select behavior, meaning you can pr
     "version": 1
 }
 ```
+
+## Text Inputs
+
+Text inputs are an interactive component that render on modals. They can be used to collect short-form or long-form text.
+
+###### Text Input Example
+
+```json
+// this is a modal
+{
+  "title": "My Cool Modal",
+  "custom_id": "cool_modal",
+  "components": [{
+    "type": 1,
+    "components": [{
+      "type": 4,
+      "custom_id": "name",
+      "label": "Name",
+      "style": 1,
+      "min_length": 1,
+      "max_length": 4000,
+      "placeholder": "John",
+      "required": true
+    }]
+  }]
+}
+```
+
+###### Text Input Structure
+
+| Field        | Type                                                                                    | Description                                                      |
+| ------------ | --------------------------------------------------------------------------------------- | ---------------------------------------------------------------- |
+| type         | integer                                                                                 | `4` for a text input                                             |
+| custom_id    | string                                                                                  | a developer-defined identifier for the input, max 100 characters |
+| style        | [Text Input Style](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/text-inputs-text-input-styles) | the style of this input                                          |
```

Comment:
shouldn't that type be an integer? like the buttons 
![image](https://user-images.githubusercontent.com/79790819/153052786-c7bddf77-46c6-4fb3-90e4-c355c5acf0c4.png)